Automattic/jetpack
# VideoPress: create endpoints to fetch and update VideoPress settings

Relates to #26938.

#### Changes proposed in this Pull Request:

* Added endpoint to read the current value for the VideoPress settings; currently, the only supported setting is the site privacy one
* Added endpoint to update the value of the settings; currently, the only supported setting is the site privacy one

#### Other information:


#### Jetpack product discussion



N/A.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Use a site with VideoPress installed
* To read the current value for the privacy information, do a `GET` request to `yoursite.com/wp-json/videopress/v1/settings`, for example:

```
curl -X GET -u "user:pass" http://yoursite.com/wp-json/videopress/v1/settings | jq
```

* The expected response is similar to this:

```
{
    "videopress_videos_private_for_site": false
}
```

* To update the value for the privacy information, do a `PUT` request to `yoursite.com/wp-json/videopress/v1/settings` setting the value for the `videopress_videos_private_for_site` field, for example:

```
curl -X PUT -u "user:pass" http://yoursite.com/wp-json/videopress/v1/settings -H 'Content-Type: application/json' -d '{"videopress_videos_private_for_site":true}'
```

* The expected response is similar to this:

```
{
   "code":"success",
   "message":"VideoPress privacy setting updated successfully.",
   "data":200
}
```

Notes:

* Make sure you have some [API auth mechanism](https://github.com/WP-API/Basic-Auth/blob/master/basic-auth.php) in place
